controllers/nnf_workflow_controller.go

```diff
@@ -884,6 +895,90 @@ func (r *NnfWorkflowReconciler) finishPreRunState(ctx context.Context, workflow
 	return nil, nil
 }
 
+func (r *NnfWorkflowReconciler) createOrUpdateDaemonSetIfNecessary(ctx context.Context, workflow *dwsv1alpha1.Workflow) error {
```

Comment:
This should move to the _helpers.go file so this file stays just start/finish logic

Reply:
Moved.